libnavigation-ui/src/main/java/com/mapbox/navigation/ui/route/NavigationMapRoute.java

```diff
@@ -290,8 +290,9 @@ public void showAlternativeRoutes(boolean alternativesVisible) {
    * @param navigation to add the progress change listener
    */
   public void addProgressChangeListener(MapboxNavigation navigation) {
+    removeProgressChangeListener(this.navigation);
```

Comment:
although this logic is not in the legacy code. But @langsmith has a good point that if you refer to the jdoc of method `removeProgressChangeListener`, I think it's a nice to have logic to prevent the potential leak.

@cafesilencio / @abhishek1508 / @Guardiola31337  , wdyt?

` ``java
/**
   * Should be called if {@link NavigationMapRoute#addProgressChangeListener(MapboxNavigation)} was
   * called to prevent leaking.
   *
   * @param navigation to remove the progress change listener
   */
  public void removeProgressChangeListener(MapboxNavigation navigation) {

Reply:
given that we are not likely to have two `MapboxNavigation`, it might be OK to not have this. 🤔

Reply:
Yeah, not sure if necessary - at least we didn't have the need in the legacy, so I believe we should be good without it.

Reply:
just removed!